spec/rspec/rails/example/view_example_group_spec.rb

```diff
@@ -248,19 +248,17 @@ def _view; end # Stub method
       end
 
       it 'is accessible to hooks', pending: pending_only_on_ruby_22_rails_52 do
-        with_isolated_config do
-          run_count = 0
-          RSpec.configuration.before(:each, type: :view) do
-            allow(view).to receive(:render) { :value }
-            run_count += 1
-          end
-          group = RSpec::Core::ExampleGroup.describe 'a view', type: :view do
-            specify { expect(true).to eq true }
-          end
-          group.run(failure_reporter)
-          expect(failure_reporter.exceptions).to eq []
-          expect(run_count).to eq 1
+        group = RSpec::Core::ExampleGroup.describe 'a view', type: :view do
+          specify { true }
+        end
+        run_count = 0
+        group.before(:each) do
+          allow(view).to receive(:render) { :value }
+          run_count += 1
         end
+        group.run(failure_reporter)
+        expect(failure_reporter.exceptions).to eq []
```

Comment:
By the way, if I add some nonsense to the hook, `failure_reporter` doesn't seem to care:
```ruby
          RSpec.configuration do
            before(:each) do
              allow(sl).to receive(:renddlfkr) { :value }
              run_count += 1
            end
          end
```
and `        expect(failure_reporter.exceptions).to eq []` is passing just fine.

So this trick doesn't prevent us from not having access to the `view`.

Does it make sense to keep it anyway @JonRowe ?

Reply:
Taking my words back.
If you comment out the whole content of `def setup` method in SetupAdnTeardownAdapter you'll get:
```diff
       -[]
       +[#<NoMethodError: undefined method `_helpers' for nil:NilClass>]
```
failure, which is much more explanatory than a `0` vs `1` failure messge.

Reply:
If you add a `fail` to the hook in the example you'll get a `RuntimeError` in the array etc, a nonsense allow statement is not an error of course, as it just allows something to be called.

It does highlight that the spec is a bit strange, the allow is not itself used, but instead is demonstrating that you can access the view in the hook, this could be improved by changing to this:

```
      # This is a regression test from rspec/rspec-rails#833
      it 'is accessible to configuration hooks', pending: pending_only_on_ruby_22_rails_52 do
        with_isolated_config do
          run_count = 0
          RSpec.configuration.before(:each, type: :view) do
            # `view` is provided from the view example type, and serves to demonstrate this
            # hook is run in the correct context.
            expect(view).to receive(:render) { :value }
            run_count += 1
          end
          group = RSpec::Core::ExampleGroup.describe 'a view', type: :view do
            specify { expect(view.render).to eq :value }
          end
          group.run(failure_reporter)
          expect(failure_reporter.exceptions).to eq []
          expect(run_count).to eq 1
        end
      end
```

Reply:
This is exactly what I've ended up with. But thanks for you time anyway!